File: Java/Topico_03/src/Banco.java
import java.util.*;

class Conta {
    private int numeroConta;
    private String nomeCliente;
    private double saldo;
    private int senha;

    public Conta(int numeroConta, int senha, String nomeCliente, double saldo) {
        this.numeroConta = numeroConta;
        this.senha = senha;
        this.nomeCliente = nomeCliente;
        this.saldo = saldo;
    }

    public int getNumeroConta() {
        return numeroConta;
    }

    public String getnomeCliente() {
        return nomeCliente;
    }

    public double getSaldo(int senha) {
        if (senha == this.senha) {
            return saldo;
        }
        return -1;
    }

    public boolean sacar(double valor, int senha) {
        if (valor > 0 && valor < this.saldo && senha == this.senha) {
            saldo -= valor;
            return true;
        }
        return false;
    }

    public boolean depositar(double valor, int senha) {
        if (valor > 0 && senha == this.senha) {
            saldo += valor;
            return true;
        }
        return false;
    }

    public boolean transferir(double valor, int senha, Conta contaDestino) {
        if (valor > 0 && valor < this.saldo && senha == this.senha){
            saldo -= valor;
            contaDestino.saldo += valor;
            return true;
        }
        return false;
    }
}

public class Banco {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        int numeroConta1 = input.nextInt();
        int senha1 = input.nextInt();
        String nomeCliente1 = input.next();
        double saldo1 = input.nextDouble();
       
        int numeroConta2 = input.nextInt();
        int senha2 = input.nextInt();
        String nomeCliente2 = input.next();
        double saldo2 = input.nextDouble();

        Conta conta1 = new Conta(numeroConta1, senha1, nomeCliente1, saldo1);
        Conta conta2 = new Conta(numeroConta2, senha2, nomeCliente2, saldo2);

        int op = 0;
        do {
            op = input.nextInt();
            switch (op) {
                case 1:
                    int senha = input.nextInt();
                    double saldo = conta1.getSaldo(senha);
                    if (saldo < 0) {
                        System.out.println("senha incorreta");
                    } else {
                        System.out.printf("%.2f\n", saldo);
                    }
                    break;
                case 2:
                    double valor = input.nextDouble();
                    senha = input.nextInt();
                    if (conta1.sacar(valor, senha)) {
                        System.out.println("saque realizado");
                    } else {
                        System.out.println("saque não realizado");
                    }
                    break;
                case 3:
                    valor = input.nextDouble();
                    senha = input.nextInt();
                    if (conta1.depositar(valor, senha)) {
                        System.out.println("depósito realizado");
                    } else {
                        System.out.println("depósito não realizado");
                    }
                    break;
                case 4:
                    String nomeCliente = input.next();
                    if (nomeCliente.equals(conta2.getnomeCliente())) {
                        valor = input.nextDouble();
                        senha = input.nextInt();
                        if (conta1.transferir(valor, senha, conta2)) {
                            System.out.println("transferência realizada");
                        } else {
                            System.out.println("transferência não realizada");
                        }
                    } else {
                        System.out.println("nenhum usuário encontrado");
                    }
                    break;
            } 
        } while (op != 5);

        input.close();
    }
}

File: C++/banco.cpp
#include <iostream>
#include <string>
using namespace std;

class Conta {
    private:
        int numeroConta, senha;
        string nomeCLiente;
        double saldo;
    
    public:
        Conta(int numeroConta, int senha, string nomeCLiente, double saldo) {
            this->numeroConta = numeroConta;
            this->senha = senha;
            this->nomeCLiente = nomeCLiente;
            this->saldo = saldo;
        }
        int getNumeroConta() {
            return numeroConta;
        }
        string getNomeCLiente() {
            return nomeCLiente;
        }
        double getSaldo(int senha) {
            if (senha == this->senha){
                return saldo;
            }
            return -1;
        }
        bool sacar(double valor, int senha) {
            if (valor > 0 && valor < this->saldo && senha == this->senha){
                saldo -= valor;
                return true;
            }
            return false;
        }
        bool depositar(double valor, int senha) {
            if (valor > 0 && senha == this->senha){
                saldo += valor;
                return true;
            }
            return false;
        }
        bool transferir(double valor, int senha, Conta contaDeDestino) {
            if (valor > 0 && valor < this->saldo && senha == this->senha){
                this->saldo -= valor;
                contaDeDestino.saldo += valor;
                return true;
            }
            return false;
        }
};

int main() {
    int numeroConta1, numeroConta2, senha1, senha2;
    string nomeCLiente1, nomeCLiente2;
    double saldo1, saldo2;
    cin >> numeroConta1 >> senha1 >> nomeCLiente1 >> saldo1;
    cin >> numeroConta2 >> senha2 >> nomeCLiente2 >> saldo2;
    Conta conta1(numeroConta1, senha1, nomeCLiente1, saldo1);
    Conta conta2(numeroConta2, senha2, nomeCLiente2, saldo2);

    cout << fixed;
    cout.precision(2);
    int op;
    int senha;
    double valor, saldo;
    string nomeCliente;

    do {
        cin >> op;
        switch (op) {
            case 1: 
                cin >> senha;
                saldo = conta1.getSaldo(senha);
                if (saldo < 0) {
                    cout << "senha incorreta" << endl;
                } else {
                    cout << saldo << endl;
                }
                break;
            case 2:
                cin >> valor >> senha;
                if (conta1.sacar(valor, senha)) {
                    cout << "saque realizado" << endl;
                } else {
                    cout << "saque n??o realizado" << endl;
                }    
                break;
            case 3:
                cin >> valor >> senha;
                if (conta1.depositar(valor, senha)) {
                    cout << "dep??sito realizado" << endl;
                } else {
                    cout << "dep??sito n??o realizado" << endl;
                }
                break;
            case 4:
                cin >> nomeCliente;
                if (nomeCliente == conta2.getNomeCLiente()) {
                    cin >> valor >> senha;
                    if (conta1.transferir(valor, senha, conta2)) {
                        cout << "transfer??ncia realizada" << endl;
                    } else {
                        cout << "transfer??ncia n??o realizada" << endl;
                    }
                } else {
                    cout << "nenhum usu??rio encontrado" << endl;
                }
                break;
        }
    } while (op != 5);

    return 0;
}